### PR TITLE
Offboarding the Morty cluster

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
Signed-off-by: greg pereira <grpereir@redhat.com>

Can I get confirmation from someone that I do not need to re-attempt backing up the `opf-datacatalog-morty` bucket into Smaug? Im not sure if it stopped due to  time out or these errors it encountered in the process: 

```
 Failed to copy `https://s3-rook-openshift-storage.apps.morty.emea.operate-first.cloud/opf-datacatalog-morty/supathak-0720121906/01_data_collection.html`. 408 Request Time-out
 
mc: <ERROR> Failed to copy `https://s3-rook-openshift-storage.apps.morty.emea.operate-first.cloud/opf-datacatalog-morty/thoth/github-pr-ttm/model-test/model-test.joblib`. Put "https://s3-openshift-storage.apps.smaug.na.operate-first.cloud/opf-datacatalog/opf-datacatalog-morty/thoth/github-pr-ttm/model-test/model-test.joblib?partNumber=19&uploadId=630e7f0fbea50f000b7582c0": read tcp 192.168.4.57:57030->46.4.65.182:443: read: connection reset by peer
```
My terminal shows it got through 9.1GB of 9.4G:
```
...metrics/bot_knowledge/operate-first/support/PullRequest.json:  9.10 GiB / 9.40 GiB  ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ░░░░░░░  4.58 MiB/s 33m57s
```

One file which would be taken care of by this PR lives in [a separate one](https://github.com/operate-first/apps/pull/2340/files) to migrate the vault backup job to smaug, courtesy of @HumairAK. As such the `vault/overlays/moc/smaug/backup-job/secret.yaml` file will not be touched in this PR. This PR should be reviewed in tandem with #2340, otherwise content would appear partially changed.

We aim to get this merged tomorrow. Please TAL:
/assign @rbo
/assign @HumairAK 
/assign @larsks 
/cc @durandom

### CODE REVIEW

**Comment by durandom** [comment]
> Can I get confirmation from someone that I do not need to re-attempt backing up the `opf-datacatalog-morty` bucket into Smaug? Im not sure if it stopped due to time out or these errors it encountered in the process:

I think it's ok to ignore some copy errors. I would guess all our really important information is in git anyways.

File: kfdefs/overlays/moc/smaug/trino/hive-metastores/kustomization.yaml [hunk 1]
```diff
@@ -4,5 +4,4 @@ kind: Kustomization
 namespace: odh-trino
 resources:
   - data-science_general
```
Comment (HumairAK) [inline kfdefs/overlays/moc/smaug/trino/hive-metastores/kustomization.yaml:6]: keep kfdefs/overlays/moc/smaug/trino contents for now, but make a follow up issue to replace the trino morty catalog with a catalog backed by a bucket in smaug

File: acm/overlays/moc/infra/managedclusters/kustomization.yaml [hunk 2]
```diff
@@ -8,5 +8,4 @@ resources:
 - osc-cl1.yaml
 - osc-cl2.yaml
 - smaug.yaml
-- morty.yaml
```
Comment (HumairAK) [inline acm/overlays/moc/infra/managedclusters/kustomization.yaml:11]: remember to also remove this in acm as well

File: cluster-scope/overlays/prod/moc/smaug/imageregistry.operator.openshift.io/configs/cluster/config_patch.yaml [hunk 3]
```diff
@@ -1,13 +0,0 @@
...
-kind: Config
-metadata:
-  name: cluster
-spec:
-  storage:
-    $patch: replace
-    managementState: Unmanaged
-    s3:
-      bucket: openshift-image-registry-bucket
-      regionEndpoint: "https://s3-rook-openshift-storage.apps.morty.emea.operate-first.cloud/"
```
Comment (HumairAK) [inline cluster-scope/overlays/prod/moc/smaug/imageregistry.operator.openshift.io/configs/cluster/config_patch.yaml:12]: smaug cluster needs an alternative bucket for image registry

Comment (Gregory-Pereira) [reply cluster-scope/overlays/prod/moc/smaug/imageregistry.operator.openshift.io/configs/cluster/config_patch.yaml:12]: bucket already exists here: https://github.com/operate-first/apps/tree/mastercluster-scope/overlays/prod/moc/smaug/objectbucketclaims/openshift-image-registry-bucket.yaml. Adding the config-patch back but instead pointing it to the existing smaug bucket.

File: kfdefs/overlays/moc/smaug/trino/hive-metastores/kustomization.yaml (hunk 1 above)
Comment (Gregory-Pereira) [reply kfdefs/overlays/moc/smaug/trino/hive-metastores/kustomization.yaml]: I have made this into a follow on issue: https://github.com/operate-first/apps/issues/2352

**Review by tumido — COMMENTED** [review]
There are couple of Morty references left in Vault backup job and Trino setup.

- https://github.com/operate-first/apps/blob/dd6daa14bfb292e5e86c0376de93be0f495339ab/vault/overlays/moc/smaug/backup-job/task.yaml#L48
- https://github.com/operate-first/apps/blob/dd6daa14bfb292e5e86c0376de93be0f495339ab/vault/overlays/moc/smaug/backup-job/secret.yaml#L10
- https://github.com/operate-first/apps/blob/dd6daa14bfb292e5e86c0376de93be0f495339ab/kfdefs/overlays/moc/smaug/trino/secrets/s3buckets.enc.yaml#L19
- https://github.com/operate-first/apps/blob/dd6daa14bfb292e5e86c0376de93be0f495339ab/kfdefs/overlays/moc/smaug/trino/hive-metastores/kustomization.yaml#L7
- https://github.com/operate-first/apps/blob/master/kfdefs/overlays/moc/smaug/trino/configs/catalogs/data_science_general_morty.properties

Otherwise this looks great!

**Comment by Gregory-Pereira** [comment]
Great catch on these Tom, I have addressed the first 3 and put them in a separate commit, as they were supposed to be dealt with in [this pr](https://github.com/operate-first/apps/pull/2340) but were not.

As for the other two...
> * https://github.com/operate-first/apps/blob/dd6daa14bfb292e5e86c0376de93be0f495339ab/kfdefs/overlays/moc/smaug/trino/hive-metastores/kustomization.yaml#L7
> * https://github.com/operate-first/apps/blob/master/kfdefs/overlays/moc/smaug/trino/configs/catalogs/data_science_general_morty.properties

Humair suggested that these changes would be out of scope ([see his comment](https://github.com/operate-first/apps/pull/2341#discussion_r959672436)) of this PR. I have a separate [follow-on issue](https://github.com/operate-first/apps/issues/2352) for tracking this. Would you still like me to address this in this PR?

**Comment by tumido** [comment]
@Gregory-Pereira that sounds good! I don't mind if the `data_science_general_morty` gets dealt with in a follow up PR or this one.. both is fine. We just need to remember to do it. :slightly_smiling_face:

**Review by tumido — APPROVED** [review]
/lgtm
/approve

/hold

app of apps is probably not set to prune resources, please watch ArgoCD and enforce resource pruning when syncing.

**Pull request merged**